zsh-history-filter.plugin.zsh

```diff
@@ -1,16 +1,43 @@
 export HISTORY_FILTER_VERSION="0.2.0"
 
-function _history_filter() {
+# overwrite the history file so that it
+# retro-actively applies the currently set filters
+function rewrite_history() {
+    local new_history="$HISTFILE.bak"
+    local excluded=0
+
+    cat $HISTFILE | while read entry; do
+        if ! _matches_filter "$entry"; then
```

Comment:
this doesnt match perfectly on the actual command because an entry in the history file is of the form

```
: 1546537882:30;rewrite_history
```